RELEASE.md

```diff
@@ -57,7 +60,7 @@ Please follow the established format:
 
 ## Bug fixes and other changes
 
-- Change route name from `runsList` to `experiment-tracking`. (#820)
+- Change route name from `runsList` to `experiment tracking`. (#820)
```

Comment:
Wasn't this right before? e.g. the URL is https://demo.kedro.org/experiment-tracking with the `-`.

Reply:
That shouldn't be there. Thanks.